main/src/ca/uwaterloo/flix/language/phase/typer/ConstraintGen.scala

```diff
@@ -743,11 +743,13 @@ object ConstraintGen {
 
         (resTpe, resEff)
 
-      case Expr.InvokeMethod2(exp, name, exps, tvar, evar, loc) =>
+      case Expr.InvokeMethod2(exp, name, exps, mvar, tvar, evar, loc) =>
         val (tpe, eff) = visitExp(exp)
         val (tpes, effs) = exps.map(visitExp).unzip
-        c.unifyType(tvar, Type.mkApply(Type.Cst(TypeConstructor.MethodReturnType(name.name, exps.length), loc), tpe :: tpes, loc), loc)
-        c.unifyType(evar, Type.mkUnion(Type.IO :: eff :: effs, loc), loc)
+        val t = Type.Cst(TypeConstructor.MethodReturnType(name.name, exps.length), loc)
+        val v = Type.IO :: eff :: effs
+        c.unifyType(tvar, Type.mkApply(t, tpe :: tpes, loc), loc)
+        c.unifyType(evar, Type.mkUnion(v, loc), loc)
         val resTpe = tvar
```

Comment:
We need one more constraint here. We need some new method on c so we can say:

```
c.addJvmEqConstraint(mvar, tpe, name, tpes, ...)
```

or something like that.